### ISSUE TITLE
Can it work with Spark SQL directly?

### ISSUE THREAD

**Issue by darouwan** [lightcopy/parquet-index#82]
I mean the SQL language in Spark SQL, rather than spark sql api. 
For example:
Dataset<Row> sqlDF =
  spark.sql("SELECT * FROM parquet.`examples/src/main/resources/users.parquet`");

**Comment by sadikovi** [comment]
Unfortunately not, this would require changes in Spark SQL parser, which at the time, I did not know how to inject. I will have a look later to see if there is a way now to make it work.

**Comment by darouwan** [comment]
Thanks!

**Comment by arunbhat** [comment]
It would be great if this works through the Thrift server. Especially when multiple tables are joined in SQL. I think the [SparkExecuteStatementOperation.scala](https://github.com/apache/spark/blob/branch-2.4/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala#L232) might need to be updated to support this.

**Comment by sadikovi** [comment]
It is tricky to do, because you would have to add rules or run custom code and modify the parser to create index or add hints to use it, when starting Spark. Right now, it is not feasible.

**Comment by arunbhat** [comment]
Yep. definitely tricky. Also will have to take care of joins and multiple tables in the SQL. Have you tested the indexing with joins? I tried and performance seems to be the same as without joins. For a single table, the performance improvement is about 35-40% based on a simple use case where a table has 60+ million rows and the filter returns about 250K rows.

**Comment by sadikovi** [comment]
It should work. Depends on how selective predicate is. The fewer rows you expect the more data it filters out.
